Update monitoring troubleshooting (#807)
harvester/docs: docs/troubleshooting/monitoring.md

### Related Issue

https://github.com/harvester/harvester/issues/5505

> ## Some PODs of `rancher-monitoring` addon are instantly killed

Reviewer: ```suggestion
## Some `rancher-monitoring` Add-On Pods Are Abruptly Terminated
```

### Issue Description

After enabling the `rancher-monitoring` addon, the `prometheus, alertmanager, grafana` related pods are keeping killed shortly after they are up.

```
$ kubectl -n cattle-monitoring-system get pods,svc,ep,deploy,pvc,sts,prometheus,alertmanager | grep -E 'stateful|deploy'

deployment.apps/rancher-monitoring-grafana              0/0     0            0           7h52m
deployment.apps/rancher-monitoring-kube-state-metrics   1/1     1            1           7h52m
deployment.apps/rancher-monitoring-operator             1/1     1            1           7h52m
deployment.apps/rancher-monitoring-prometheus-adapter   1/1     1            1           7h52m
statefulset.apps/alertmanager-rancher-monitoring-alertmanager   0/0     7h52m
statefulset.apps/prometheus-rancher-monitoring-prometheus       0/0     7h52m
```

> When check the `prometheus` pod log, it might have the keyword `level=warn msg="Received SIGTERM, exiting gracefully..."`.

Reviewer: ```suggestion
The `prometheus` pod logs contain the message `level=warn msg="Received SIGTERM, exiting gracefully..."`.

Example:
```

...
```

### Root Cause

> When [storage network](../advanced/storagenetwork.md#harvester-storage-network-setting) setting is changed, Harvester controller checks and waits the attached volumes to be detached, then applies the new configuration. Meanwhile, the controller kills `prometheus`, `grafana` and `alertmanager` related pods automatically as those pods are using volumes to store data. Normally this process consists for a short while.

Reviewer: ```suggestion
When you make changes to the [storage-network](../advanced/storagenetwork.md#harvester-storage-network-setting) setting, the Harvester controller waits for the attached volumes to be detached before applying the changes. In addition, the controller automatically terminates the pods related to Prometheus, Alertmanager, and Grafana because those pods use volumes to store data.
```

> If the [storage network](../advanced/storagenetwork.md#harvester-storage-network-setting) change is unfortunately blocked by some attached volumes, and the user or the monitoring-operator tries to bring up the `rancher-monitoring`, then the pods are killed by the controller quickly, the aforementioined phenomenon is observed.

Reviewer: ```suggestion
This process usually takes a short time to complete, but can be disrupted when the following occur:

- Attached volumes prevent the Harvester controller from applying the changes to the setting.
- A user or the `monitoring-operator` attempts to use the `rancher-monitoring` add-on.
- The Harvester controller terminates the pods.
```

Reviewer: The draft was unclear. Let me know if I misunderstood anything, particularly this part: "the user or the monitoring-operator tries to bring up the `rancher-monitoring`"
